pkg/agent/conversation.go

```diff
@@ -546,6 +551,12 @@ func (c *Agent) Run(ctx context.Context, initialQuery string) error {
 				if streamedText != "" {
 					c.addMessage(api.MessageSourceModel, api.MessageTypeText, streamedText)
 				}
+
+				if pct, ok := gollm.ContextPercentRemaining(c.Model, usageMetadata); ok {
+					// c.addMessage(api.MessageSourceAgent, api.MessageTypeText, fmt.Sprintf("Context remaining: %.1f%%", pct))
+					c.addMessage(api.MessageSourceAgent, api.MessageTypeUsageMetadata, api.UsageMetadata{ContextPercentRemaining: pct})
```

Comment:
I think it will make more sense to just make this (contextRemaining) an attribute of the session object and update it whenever session's context is being updated. UI has access to the session object and we used that to communicate `AgentState` attribute.